Secret-agency-service/src/main/java/cz/fi/muni/pa165/secretagency/service/GenericServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+package cz.fi.muni.pa165.secretagency.service;
+
+import cz.fi.muni.pa165.secretagency.dao.GenericDao;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+/**
+ * Implementation of generic service. It provides general method implementations.
+ * Every other Service implementation should extend this one.
+ *
+ * This class needs specific dao as an second typed argument to be able to provide
+ *    specific dao methods for services which extend this one.
+ *
+ * @param <Entity> Entity
+ *
+ * 2author Jan Pavlu (487548)
+ */
+@Service
+public class GenericServiceImpl<Entity, Dao extends GenericDao<Entity>> implements GenericService<Entity> {
+
+    @Autowired
+    protected Dao dao;
```

Comment:
tu mi to rve `Could not autowire. No beans of 'Dao' type found.` u tebe je to vcajku?

Reply:
Já jsem se k tomu, abych to někde konkrétně použil, ještě nedostal. Teď jdu na to.

Reply:
ok, to rve i bez pouziti 😄